### PR TITLE
[NFC][SYCL] Use get_elem_type_t instead of TryToGetElementType<T>::type

### PR DESCRIPTION
**Pull request by aelovikov-intel**

### CODE REVIEW

**Review by bso-intel — APPROVED** [review]

File: sycl/include/sycl/detail/generic_type_traits.hpp [hunk 1]
```diff
@@ -645,6 +633,18 @@ template <typename T, typename Enable = void> struct RelConverter {
...
+// exist, otherwise T.
+template <typename T> class TryToGetElementType {
+  static T check(...);
+  template <typename A> static typename A::element_type check(const A &);
+  template <typename A> static typename A::value_type check(const A &);
+
+public:
+  using type = decltype(check(T()));
+  static constexpr bool value = !std::is_same_v<T, type>;
+};
```
Comment (steffenlarsen) [inline sycl/include/sycl/detail/generic_type_traits.hpp:646]: Why do we keep `TryToGetElementType`?

Comment (aelovikov-intel) [reply sycl/include/sycl/detail/generic_type_traits.hpp:646]: It has a single remaining use just below at line 650. Note that it's `::value`, not `::type` so I can't do the same replacement as in other places of this PR.

Comment (steffenlarsen) [reply sycl/include/sycl/detail/generic_type_traits.hpp:646]: I see. I personally think `value` in this is unintuitive, so I would prefer we replace it, if nothing else then by just replacing the use with `!std::is_same_v<T, get_elem_type_t<T>>`.

**Review by steffenlarsen — APPROVED** [review]
I am fine with my comment being addressed separately, but I also don't think it necessarily muddies this PR too much to just inline the one remaining use case.

File: sycl/include/sycl/detail/generic_type_traits.hpp (hunk 1 above)
Comment (aelovikov-intel) [reply sycl/include/sycl/detail/generic_type_traits.hpp:526]: It's not immediately obvious if that would be an NFC or not, so I won't be doing it here.

**Pull request merged**